infra/src/main/java/com/redhat/service/smartevents/infra/models/filters/BaseFilter.java

```diff
@@ -49,5 +49,5 @@ public String getType() {
 
     @NotNull
     @JsonIgnore
-    public abstract T getValue();
+    public abstract Object getValue();
```

Comment:
I am trying to understand why you made this change?

It seems to move filters further away from the "typed" form I thought we were moving towards.

Was it just to simplify the generics?

Reply:
The generic type in `BaseFilter` was used only for the return type of the `getValue` method, but Java allows to override a method specifying a stricter return type than the superclass without the need of generics (i.e. return type in the subclass is a subclass of the return type in the superclass).

Thus, using plain `Object` in `BaseFilter` achieves the same goal and also eliminates all the warnings that SonarCloud was throwing because `BaseFilter` was used *everywhere* in the codebase except in one place *without the generic type specified* (i.e. `BaseFilter` instead of `BaseFilter<...>`). SonarCloud warnings are the main reason why I did this.